background/script.js

```diff
@@ -343,15 +364,29 @@ async function assumeRole(
   SessionDuration
 ) {
   // Set the fetched STS keys from the SAML response as credentials for doing the API call
-  let clientconfig = {
-    region: "us-east-1", // region is mandatory to specify, but ignored when using global endpoint
-    useGlobalEndpoint: true,
-    credentials: {
-      accessKeyId: AccessKeyId,
-      secretAccessKey: SecretAccessKey,
-      sessionToken: SessionToken,
-    },
-  };
+  var RolePattern = /arn:(aws|aws-us-gov|aws-cn):iam:[^:]*:[0-9]+:role\/[^,]+/i;
+  AWSPartition = roleArn.match(RolePattern)[1];
+  if (AWSPartition == "aws-cn") {
+    let clientconfig = {
+      region: "cn-north-1",
+      credentials: {
+        accessKeyId: AccessKeyId,
+        secretAccessKey: SecretAccessKey,
+        sessionToken: SessionToken,
+      },
+    };
+  } else {
+    let clientconfig = {
+      region: "us-east-1", // region is mandatory to specify, but ignored when using global endpoint
+      useGlobalEndpoint: true,
+      credentials: {
+        accessKeyId: AccessKeyId,
+        secretAccessKey: SecretAccessKey,
+        sessionToken: SessionToken,
+      },
+    };
+  }
+
   // AWS SDK is a module exorted from a webpack packaged lib
   // See 'library.name' in webpack.config.js
   const client = new webpacksts.AWSSTSClient(clientconfig);
```

Comment:
We should do one adjustment.

```
ReferenceError: clientconfig is not defined
    at assumeRole (script.js:392:46)

```

Soit means thath clientconfig is block scoped and his reference not comming to the `const client = new webpacksts.AWSSTSClient(clientconfig);`

Pls use var or instead or make the default the aws credentials and if the AWSPartition is "aws-cn" then rewrite it.